packages/ethers-5/src/client.ts

```diff
@@ -0,0 +1,788 @@
+import type {
+  CaipAddress,
+  CaipNetwork,
+  CaipNetworkId,
+  ConnectionControllerClient,
+  Connector,
+  LibraryOptions,
+  NetworkControllerClient,
+  PublicStateControllerState,
+  Token
+} from '@web3modal/scaffold'
+import { Web3ModalScaffold } from '@web3modal/scaffold'
+import {
+  ADD_CHAIN_METHOD,
+  COINBASE_CONNECTOR_ID,
+  EIP6963_ANNOUNCE_EVENT,
+  EIP6963_CONNECTOR_ID,
+  EIP6963_REQUEST_EVENT,
+  INJECTED_CONNECTOR_ID,
+  NAMESPACE,
+  VERSION,
+  WALLET_CONNECT_CONNECTOR_ID,
+  WALLET_ID,
+  ConnectorExplorerIds,
+  ConnectorImageIds,
+  ConnectorNamesMap,
+  ConnectorTypesMap,
+  NetworkBlockExplorerUrls,
+  NetworkImageIds,
+  NetworkNames,
+  NetworkRPCUrls,
+  networkCurrenySymbols
+} from '@web3modal/utils'
+import type EthereumProvider from '@walletconnect/ethereum-provider'
+import {
+  caipNetworkIdToNumber,
+  getEthersCaipDefaultChain,
+  getCaipTokens,
+  hexStringToNumber
+} from '@web3modal/utils'
+import type { Address, ProviderType } from './utils/types.js'
+import { ethers, utils } from 'ethers'
+import { ProviderController } from './store/index.js'
+
+// -- Types ---------------------------------------------------------------------
+export interface Web3ModalClientOptions extends Omit<LibraryOptions, 'defaultChain' | 'tokens'> {
```

Comment:
I think a lot of these types are now repeated in wagmi and ethers? We could move them to utilities as well?

Make sure to keep utilities organised i.e. split types/presets/constants etc into their own files